### PR TITLE
Add a nayduck/python test for switching from memtries to disktries

### PR DESCRIPTION
**Pull request by tayfunelmas**
This tests, in case needed due to an issue with memtries, whether nodes can switch from memtries to disktries and switch back to memtries later.

### CODE REVIEW

**Comment by Longarithm** [comment]
Would you like to try robust TestLoop? :)

**Comment by tayfunelmas** [comment]
> Would you like to try robust TestLoop? :)

Yeah TestLoop is way more stable (seeing it clearly as I run nayduck now :P), but I need a test that exercises the node fully end-to-end and allows me to restart the nodes multiple times.

File: pytest/tests/sanity/memtrie_disktrie_switch.py [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+                    self.next_nonce(from_account_key), last_block_hash)
+                result = self.rpc_node.send_tx(payment_tx)
+                assert 'result' in result and 'error' not in result, (
+                    'Expected "result" and no "error" in response, got: {}'.
+                    format(result))
+                logger.debug("Transfer: {}".format(result))
+                tx_hash = result['result']
+                self.txs.append((from_account_key.account_id, tx_hash))
+            elif len(self.keys) > 10 and random.random() < 0.5:
+                # Do some flat storage reads, but only if we have enough keys populated.
```
Comment (pugachAG) [inline pytest/tests/sanity/memtrie_disktrie_switch.py:190]: nit: I suggest replacing flat storage with just storage here and at line 205: for disktrie flat storage is just an implementation detail and for memtrie it is not even relevant

**Review by pugachAG — APPROVED** [review]
LGTM

**Pull request merged**